src/io/bnx.rs

```diff
@@ -0,0 +1,1486 @@
+//! Structs amd trait to read files in BNX format.
+//!
+//! Note: This module only supports version v1.2 of the BNX format for now.
+//!
+//! # Example
+//!
+//! ## Reader
+//!
+//! In this example, we parse a BNX file, iterate over its records. and output some statistics.
+//!
+//! ```rust
+//! use bio::io::bnx;
+//!
+//! let path = "tests/resources/valid_input.bnx";
+//! let mut reader = bnx::Reader::from_path(&path).unwrap();
+//!
+//! let mut num_molecules = 0;
+//! let mut num_labels_channel_1 = 0;
+//! let mut num_labels_channel_2 = 0;
+//!
+//! while let Some(Ok(mut record)) = reader.next() {
+//!     num_molecules += 1;
+//!     num_labels_channel_1 += record.channel_1().len() - 1;
+//!     num_labels_channel_2 += record.channel_2().len() - 1;
+//! }
+//!
+//! println!("Number of molecules: {}", num_molecules);
+//! println!("Number of labels in channel 1: {}", num_labels_channel_1);
+//! println!("Number of labels in channel 2: {}", num_labels_channel_2);
+//! ```
+//!
+//! ## Container
+//!
+//! If feasible, we can also build a container over the content of a BNX file.
```

Comment:
The meaning and purpose of a container is not clear to the reader here. Needs some explanation.

Reply:
I think it can be useful if you want the whole file read in at once.

Reply:
This may be relevant if the BNX is unsorted or one may need to access the entries out-of-order.

Reply:
Indeed, but this should be mentioned in the comment.